### PR TITLE
CSharp/aspnetcore update

### PR DESCRIPTION
**Pull request by mikeharder**
- Update framework and packages to 1.1
- Remove dependency on net451 and mono
- Update PostgreSQL connection string for latest npgsql
  - Prepared statements are now persisted by default, so the PersistPrepared option has been removed.
  - npgsql/npgsql#1237
- Make shell scripts executable
- Change scripts to use `dotnet Benchmarks.dll`
  - Scripts currently use `dotnet run` which is intended for iterative development rather than production.
  - Should not cause a measurable change in performance.
- Add DB scenarios to README.md

@pranavkm: Can you please review the changes to project.json?  I was updating the packages from 1.0 to 1.1, and removing the floating versions.  npgsql still needs to float since we are using CI builds from MyGet.  Once npgsql 3.2 beta is released on NuGet, we can remove this float as well.

@dougbu: Can you please review the other changes?

CC: @davidfowl, @DamianEdwards, @nathana1

### CODE REVIEW

File: frameworks/CSharp/aspnetcore/Benchmarks/appsettings.postgresql.json [hunk 1]
```diff
@@ -1,4 +1,4 @@
 ﻿{
-  "ConnectionString": "Server={db_server_placeholder};Database=hello_world;User Id=benchmarkdbuser;Password=benchmarkdbpass;Maximum Pool Size=1024;NoResetOnClose=true;PersistPrepared=true",
+  "ConnectionString": "Server={db_server_placeholder};Database=hello_world;User Id=benchmarkdbuser;Password=benchmarkdbpass;Maximum Pool Size=1024;NoResetOnClose=true",
```
Comment (dougbu) [inline frameworks/CSharp/aspnetcore/Benchmarks/appsettings.postgresql.json]: Just curious (no change requested): Does it hurt anything to be explicit about persisting prepared statements?

File: frameworks/CSharp/aspnetcore/Benchmarks/project.json [hunk 2]
```diff
@@ -12,43 +12,38 @@
...
+    "Microsoft.Extensions.Configuration.EnvironmentVariables": "1.1.0",
     "Npgsql": "3.2.0-*",
-    "Npgsql.EntityFrameworkCore.PostgreSQL": "1.0.2-*" 
+    "Npgsql.EntityFrameworkCore.PostgreSQL": "1.1.0"
   },
   "frameworks": {
-    "netcoreapp1.0": {
-      "imports": [ "portable-net451+win8", "dnxcore50" ],
+    "netcoreapp1.1": {
+      "imports": "dnxcore50",
```
Comment (dougbu) [inline frameworks/CSharp/aspnetcore/Benchmarks/project.json]: What still needs `imports` in 1.1.0?

File: frameworks/CSharp/aspnetcore/Benchmarks/project.json [hunk 3]
```diff
@@ -12,43 +12,38 @@
...
   },
   "tools": {
     "Microsoft.EntityFrameworkCore.Tools": {
-      "version": "1.0.0-*",
-      "imports": [
-        "dnxcore50",
-        "portable-net452+win81"
-      ]
+      "version": "1.1.0-preview4-final"
     }
```
Comment (dougbu) [inline frameworks/CSharp/aspnetcore/Benchmarks/project.json]: nit: Might as well squish this down to `"Microsoft.EntityFrameworkCore.Tools": "1.1.0-preview4-final"`.

File: frameworks/CSharp/aspnetcore/run-linux.sh [hunk 4]
```diff
@@ -11,6 +11,6 @@ cd Benchmarks
 cp appsettings.postgresql.json appsettings.json
 sed -i 's|{db_server_placeholder}|'"${DBHOST}"'|g' appsettings.json
 dotnet restore
-dotnet build -c Release -f netcoreapp1.0
+dotnet build -c Release
```
Comment (dougbu) [inline frameworks/CSharp/aspnetcore/run-linux.sh]: Why this change (versus updating the version)? No reason to build more than the DLL that's about to execute.

File: frameworks/CSharp/aspnetcore/run-windows.ps1 [hunk 5]
```diff
@@ -4,5 +4,5 @@ $scenarios = (-split $scenarios) -join ","
 
 cd Benchmarks
 dotnet restore
-dotnet build -c Release -f netcoreapp1.0
-Start-Process -NoNewWindow dotnet "run -c Release server.urls=http://*:8080 server=$server threadCount=1 NonInteractive=true scenarios=$scenarios"
+dotnet build -c Release
```
Comment (dougbu) [inline frameworks/CSharp/aspnetcore/run-windows.ps1]: Again, update to `netcoreapp1.1` but otherwise leave this line alone.

File: frameworks/CSharp/aspnetcore/Benchmarks/appsettings.postgresql.json (hunk 1 above)
Comment (mikeharder) [inline frameworks/CSharp/aspnetcore/Benchmarks/appsettings.postgresql.json]: The "PersistPrepared" option is no longer valid in the connection string.  Instead, prepared statements are always persisted.  We are depending on an npgsql feature still under development, which is why there was a breaking change.  See https://github.com/npgsql/npgsql/issues/1237 for details.

File: frameworks/CSharp/aspnetcore/Benchmarks/project.json (hunk 2 above)
Comment (mikeharder) [inline frameworks/CSharp/aspnetcore/Benchmarks/project.json]: I will try removing this import.  It's probably not needed, though "dotnet new" still adds this import, so I left it.

File: frameworks/CSharp/aspnetcore/Benchmarks/project.json (hunk 3 above)
Comment (mikeharder) [inline frameworks/CSharp/aspnetcore/Benchmarks/project.json]: OK

File: frameworks/CSharp/aspnetcore/run-linux.sh (hunk 4 above)
Comment (mikeharder) [inline frameworks/CSharp/aspnetcore/run-linux.sh]: I think it's unnecessary, since there's only one framework in project.json, since I removed `net451`.  If there's any reason to still specify the framework to `dotnet build` when there's only one in project.json, I can add it back.

File: frameworks/CSharp/aspnetcore/Benchmarks/appsettings.postgresql.json (hunk 1 above)
Comment (dougbu) [inline frameworks/CSharp/aspnetcore/Benchmarks/appsettings.postgresql.json]: 👌

File: frameworks/CSharp/aspnetcore/run-linux.sh (hunk 4 above)
Comment (dougbu) [inline frameworks/CSharp/aspnetcore/run-linux.sh]: 👌

File: frameworks/CSharp/aspnetcore/run-windows.ps1 (hunk 5 above)
Comment (dougbu) [inline frameworks/CSharp/aspnetcore/run-windows.ps1]: Fine to 🙈

**Comment by pranavkm** [comment]
csproj changes looks good to me.

**Comment by mikeharder** [comment]
Merged: https://github.com/TechEmpower/FrameworkBenchmarks/pull/2483
